### ISSUE TITLE
Issue with ng server with new project

### ISSUE THREAD

**Issue by k123009** [angular/angular-cli#2859]
Hi,
I am new to angular 2 and creating hello world program but unable to test my first angular 2 application as ng server command is executing with below logs with multiple exceptions.

Here is version information so for
angular-cli: 1.0.0-beta.18
node: 4.6.1
os: win32 x64

C:\Users\Moin\angular2_HelloWorld>ng server
*\* NG Live Development Server is running on http://localhost:4200. **
5921ms building modules
36ms sealing
1ms optimizing
1ms basic module optimization
122ms module optimization
0ms advanced module optimization
16ms basic chunk optimization
0ms chunk optimization
0ms advanced chunk optimization
0ms module and chunk tree optimization
61ms module reviving
6ms module order optimization
4ms module id optimization
5ms chunk reviving
0ms chunk order optimization
20ms chunk id optimization
48ms hashing
4ms module assets processing
124ms chunk assets processing
4ms additional chunk assets processing
0ms recording
0ms additional asset processing
140ms chunk asset optimization
1334ms asset optimization
30ms emitting
Hash: 1d89c02908e51bd759c1
Version: webpack 2.1.0-beta.25
Time: 7893ms
           Asset       Size  Chunks             Chunk Names
  main.bundle.js     2.7 MB    0, 2  [emitted]  main
styles.bundle.js    10.3 kB    1, 2  [emitted]  styles
       inline.js    5.37 kB       2  [emitted]  inline
        main.map    2.73 MB    0, 2  [emitted]  main
      styles.map    13.9 kB    1, 2  [emitted]  styles
      inline.map    5.55 kB       2  [emitted]  inline
      index.html  485 bytes          [emitted]

ERROR in ./~/webpack-dev-server/client?http://localhost:4200/
Module not found: Error: Can't resolve 'strip-ansi' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\webpack-dev-server\client'
 @ ./~/webpack-dev-server/client?http://localhost:4200/ 2:16-37
 @ multi main

ERROR in ./~/sockjs-client/lib/main.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib'
 @ ./~/sockjs-client/lib/main.js 26:10-26
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/info-receiver.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib'
 @ ./~/sockjs-client/lib/info-receiver.js 16:10-26
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/iframe-bootstrap.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib'
 @ ./~/sockjs-client/lib/iframe-bootstrap.js 14:10-26
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/websocket.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport'
 @ ./~/sockjs-client/lib/transport/websocket.js 12:10-26
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/url.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\utils'
 @ ./~/sockjs-client/lib/utils/url.js 7:10-26
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/transport.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\utils'
 @ ./~/sockjs-client/lib/utils/transport.js 5:10-26
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/info-iframe.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib'
 @ ./~/sockjs-client/lib/info-iframe.js 13:10-26
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/info-ajax.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib'
 @ ./~/sockjs-client/lib/info-ajax.js 11:10-26
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/utils/iframe.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\utils'
 @ ./~/sockjs-client/lib/utils/iframe.js 10:10-26
 @ ./~/sockjs-client/lib/iframe-bootstrap.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/ajax-based.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\lib'
 @ ./~/sockjs-client/lib/transport/lib/ajax-based.js 10:10-26
 @ ./~/sockjs-client/lib/transport/xhr-streaming.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/iframe.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport'
 @ ./~/sockjs-client/lib/transport/iframe.js 23:10-26
 @ ./~/sockjs-client/lib/transport/lib/iframe-wrap.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/sender-receiver.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\lib'
 @ ./~/sockjs-client/lib/transport/lib/sender-receiver.js 11:10-26
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/url-parse/index.js
Module not found: Error: Can't resolve 'requires-port' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\url-parse'
 @ ./~/url-parse/index.js 3:15-39
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/url-parse/index.js
Module not found: Error: Can't resolve 'querystringify' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\url-parse'
 @ ./~/url-parse/index.js 5:9-34
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/sender/xdr.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\sender'
 @ ./~/sockjs-client/lib/transport/sender/xdr.js 12:10-26
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/sender/jsonp.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\sender'
 @ ./~/sockjs-client/lib/transport/sender/jsonp.js 9:10-26
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/xhr.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\receiver'
 @ ./~/sockjs-client/lib/transport/receiver/xhr.js 9:10-26
 @ ./~/sockjs-client/lib/transport/xhr-streaming.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/eventsource.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\receiver'
 @ ./~/sockjs-client/lib/transport/receiver/eventsource.js 10:10-26
 @ ./~/sockjs-client/lib/transport/eventsource.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/jsonp.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\receiver'
 @ ./~/sockjs-client/lib/transport/receiver/jsonp.js 13:10-26
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/receiver/htmlfile.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\receiver'
 @ ./~/sockjs-client/lib/transport/receiver/htmlfile.js 12:10-26
 @ ./~/sockjs-client/lib/transport/htmlfile.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/buffered-sender.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\lib'
 @ ./~/sockjs-client/lib/transport/lib/buffered-sender.js 9:10-26
 @ ./~/sockjs-client/lib/transport/lib/sender-receiver.js
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/lib/polling.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\lib'
 @ ./~/sockjs-client/lib/transport/lib/polling.js 9:10-26
 @ ./~/sockjs-client/lib/transport/lib/sender-receiver.js
 @ ./~/sockjs-client/lib/transport/jsonp-polling.js
 @ ./~/sockjs-client/lib/transport-list.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main

ERROR in ./~/sockjs-client/lib/transport/browser/abstract-xhr.js
Module not found: Error: Can't resolve 'debug' in 'C:\Users\Moin\angular2_HelloWorld\node_modules\sockjs-client\lib\transport\browser'
 @ ./~/sockjs-client/lib/transport/browser/abstract-xhr.js 12:10-26
 @ ./~/sockjs-client/lib/transport/sender/xhr-local.js
 @ ./~/sockjs-client/lib/info-receiver.js
 @ ./~/sockjs-client/lib/main.js
 @ ./~/sockjs-client/lib/entry.js
 @ ./~/webpack-dev-server/client/socket.js
 @ ./~/webpack-dev-server/client?http://localhost:4200/
 @ multi main
Child html-webpack-plugin for "index.html":
         Asset     Size  Chunks       Chunk Names
    index.html  2.82 kB       0
webpack: bundle is now VALID.

Kindly suggest

Thank you
Muhiuddin

**Comment by esafwan** [comment]
No idea if the issue is same in windows. I had a similar error in my mac when I used node v4.6.1. The issue disappeared once I updated to v6.9.1. See the issue [here.](https://github.com/angular/angular-cli/issues/2844#event-834473253)

**Comment by knobo** [comment]
You forgot to do "npm install" in your project directory.
You forgot to do it because the tutorial does not say you need to do it.

**Comment by mmilkov** [comment]
Same here:

```
ERROR in multi main
Module not found: Error: Can't resolve 'webpack-dev-server/client?http://IP:4200/' in '/home/user/.nvm/versions/node/v6.9.1/lib/node_modules/angular-cli/models'
 @ multi main
```

:\

**Comment by p-engelkes** [comment]
Same here... Windows 10

**Comment by mmilkov** [comment]
I fixed my issue #2891 Hope this help someone. :)))

**Comment by filipesilva** [comment]
This seems to be a dependency problem as @knobo said.
